wgpu-hal/src/dx12/command.rs

```diff
@@ -1272,13 +1273,185 @@ impl crate::CommandEncoder for super::CommandEncoder {
     {
         // Implement using `BuildRaytracingAccelerationStructure`:
         // https://microsoft.github.io/DirectX-Specs/d3d/Raytracing.html#buildraytracingaccelerationstructure
-        todo!()
+        let list = self
+            .list
+            .as_ref()
+            .unwrap()
+            .cast::<Direct3D12::ID3D12GraphicsCommandList4>()
+            .unwrap();
+        for descriptor in descriptors {
+            // TODO: This is the same as getting build sizes apart from requiring buffers, should this be de-duped?
+            let mut geometry_desc;
+            let (ty, layout, inputs0, num_desc) = match descriptor.entries {
```

Comment:
This a readability nit, for tuple returns, I much prefer late initialized values. So:

```suggestion
            let ty;
            let layout;
            let inputs0;
            let num_desc;
            match descriptor.entries {
```

Then in each branch, you set `ty = blah;` `layout = blah` etc. This way you can clearly see which name is being assigned which value.